src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -28,6 +28,11 @@ public static bool IsLinuxContainerEnvironment(this IEnvironment environment)
             return !environment.IsAppServiceEnvironment() && !string.IsNullOrEmpty(environment.GetEnvironmentVariable(ContainerName));
         }
 
+        public static bool IsPreSpecializedLinuxContainer(this IEnvironment environment)
+        {
+            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(ContainerStartContext));
```

Comment:
I think it would be better to include (&& environment.IsLinuxContainerEnvironment()) to match the logic to name of the method.